src/main/java/com/emc/pravega/perf/ReaderWorker.java

```diff
@@ -46,17 +46,17 @@
      */
     public abstract void close();
 
-
     @Override
     public Void call() {
         String ret = null;
         try {
 
             for (int i = 0; (i < eventsPerWorker) &&
                     (Duration.between(StartTime, Instant.now()).getSeconds() < secondsToRun); i++) {
+                final Instant startTime = Instant.now();
                 ret = readData();
                 if (ret != null) {
-                    stats.recordTime(null, Instant.ofEpochMilli(Long.parseLong(ret.split(",")[0])), ret.length());
+                    stats.recordTime(null, startTime, ret.length());
```

Comment:
This line is changing what is being measured. Before it was measuring writer->reader latency, now it is measuring the latency of `readData()`. Those are different, I want to check if this is intended.

Reply:
yes, for now.. it will be latency of reads ; once i published depending on the inputs , we may have write another type of producer /consumers for writer->reader latencies.